### PR TITLE
Update softmaker-freeoffice

### PR DESCRIPTION
**Pull request by lenlo**
Created with `brew bump-cask-pr`.

### CODE REVIEW

**Comment by lenlo** [comment]
I forgot to mention that this version apparently represents Service Pack 1014 that was released yesterday and that everyone who uses Monterey need to update to. Does anyone know what the best way is to represent this in Homebrew? The path is apparently unchanged.

**Comment by lenlo** [comment]
I'm sorry, but how do I add a label? I don't see an option to do so anywhere. Is this only available to contributors with additional privileges?

**Comment by bevanjkay** [comment]
@neutric Do you have any ideas here? It looks like the shasum will need to be `:no_check` because the package is updated in place. But we may be able to include the build number in the version with livecheck.

**Comment by neutric** [comment]
@bevanjkay Yes to `sha256 :no_check`. This is what I came up with regarding revisions:

```
livecheck do
  url "https://www.freeoffice.com/en/download/servicepacks"
  strategy :page_match do |page|
    match = page.match(/freeoffice[._-]v?(\d+(?:\.\d+)*)\.pkg.*?Revision\s*(\d+)\s*</im)
    next if match.blank?

    "#{match[1]},#{match[2]}"
  end
end
```

**Comment by miccal** [comment]
Thank you @lenlo.

File: Casks/softmaker-freeoffice.rb [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
   homepage "https://www.freeoffice.com/"
 
   livecheck do
     url "https://www.freeoffice.com/en/download/servicepacks"
-    regex(/softmaker[._-]freeoffice[._-](\d+(?:\.\d+)*)\.pkg/i)
+    strategy :page_match do |page|
+      match = page.match(/(\d+)-(\d+)-(\d+):\s*Revision\s*(\d+)/i)
+      next if match.blank?
+
+      "#{match[1]},#{match[4]}"
```
Comment (neutric) [inline Casks/softmaker-freeoffice.rb:17]: @miccal Your regex is not going to serve us well come next year. Up until [a few months ago](https://github.com/Homebrew/homebrew-cask/pull/102583) it was still the 2018 edition of FreeOffice despite the year being 2021.

Comment (miccal) [reply Casks/softmaker-freeoffice.rb:17]: Feel free to submit a PR for a better one.

**Pull request merged**